java/jdbc/src/main/java/io/vitess/jdbc/VitessVTGateManager.java

```diff
@@ -56,7 +56,7 @@
          */
         public VTGateConnections(VitessConnection connection) {
             for (VitessJDBCUrl.HostInfo hostInfo : connection.getUrl().getHostInfos()) {
-                String identifier = getIdentifer(hostInfo.getHostname(), hostInfo.getPort(), connection.getUsername());
+                String identifier = getIdentifer(hostInfo.getHostname(), hostInfo.getPort(), connection.getUsername(), connection.getKeyspaceShard());
```

Comment:
sorry, this got conflicted with some other internal work I have going.  fixed now